pkg_resources/__init__.py

```diff
@@ -108,6 +125,19 @@
     stacklevel=2,
 )
 
+if TYPE_CHECKING:
+    from _typeshed import StrPath
+    from types import _LoaderProtocol
+
+# Type aliases
+_NestedStr = Union[str, Iterable[Union[str, Iterable["_NestedStr"]]]]
+_InstallerType = Callable[["Requirement"], Optional["Distribution"]]
+_PkgReqType = Union[str, "Requirement"]
+_EPDistType = Union["Distribution", _PkgReqType]
+_MetadataType = Optional["IResourceProvider"]
```

Comment:
These aliases come from the typeshed stub. Please lmk if you'd prefer different names